### PR TITLE
Make sure all log events are emitted before shutdown

### PR DESCRIPTION
**Pull request by mururu**
Engine emits all log events before shutdown RootAgent.

I'll send a same patch to the master branch after this PR is merged.

### CODE REVIEW

File: lib/fluent/engine.rb [hunk 1]
```diff
@@ -209,11 +209,11 @@ def run
         $log.error_backtrace
       ensure
         $log.info "shutting down fluentd"
-        shutdown
         if @log_emit_thread
           @log_event_loop_stop = true
           @log_emit_thread.join
         end
+        shutdown
```
Comment (repeatedly) [inline lib/fluent/engine.rb]: Does calling `shutdown` after log thread stop cause log inside `shutdown` is dropped?

Comment (mururu) [inline lib/fluent/engine.rb]: Yes. I wanted to make sure that the output plugin capturing events isn't terminated when log_emit_thread emit events. It seems to cause non-deterministic behaviour. But I'll try to capture more events event in `shutdown`.

Comment (repeatedly) [inline lib/fluent/engine.rb]: I see. We should dump buffered log event `log_event_queue` after shutdowned.

**Comment by repeatedly** [comment]
Merged. Wait next patch :)

**Comment by repeatedly** [comment]
Can we release new v0.12 or should wait new patch?

**Comment by mururu** [comment]
I think we can release the new version.

**Comment by mururu** [comment]
^ I'll send a new patch probably tomorrow.

**Comment by mururu** [comment]
@repeatedly Here is the next patch: https://github.com/fluent/fluentd/pull/1625. If you have time before releasing the new version, please review it.

**Pull request merged**